### PR TITLE
wolfHSM Manual (migration from wolfHSM-docs to wolfssl/documentation)

### PR DESCRIPTION
**Pull request by aidangarske**
wolfHSM documentation generation.

The wolfHSM does not yet have API documentation, so the Doxygen steps are being skipped. Only the src/*.md files are being used for the HTML and PDF manuals. 

Updated the mkdocs.yml and took out the SUMMARY.md because it is not necessary in documentations repo.

Adds new V=1 option to help with showing commands being used. Updated copyright.

### CODE REVIEW

**Comment by dgarske** [comment]
The failure is due to the existing `#17 1.500 cat: api/md/tpm2_8h.md: No such file or directory` issue. I'm going to try and fix that right now. 

I locally built `make wolfhsm` and got the error:

```
1.661 pandoc: SUMMARY.md: withBinaryFile: does not exist (No such file or directory)
```
[wolfHSM-Manual.pdf](https://github.com/user-attachments/files/15593154/wolfHSM-Manual.pdf)


Fix pushed. Attached is the wolfHSM User Manual for review.

File: wolfHSM/src/chapter03.md
Comment (bigbrett) [inline wolfHSM/src/chapter03.md:114]: spurious newline
```suggestion
*Note: A wolfHSM port comes with a reference server application that is already configured to run on the HSM core and so manual server configuration is not required.*
```

File: wolfHSM/src/chapter03.md
Comment (bigbrett) [inline wolfHSM/src/chapter03.md:133]: weird - the nested bullets are rendered as 1.a, 1.b, 1.c, despite them being written as 1.1, 1.2, 1.3. So referring to "step 1.1" in the text is broken

File: wolfHSM/src/chapter04.md
Comment (bigbrett) [inline wolfHSM/src/chapter04.md:78]:
```suggestion
### Client/Server APIs
```

File: wolfHSM/src/chapter04.md
Comment (bigbrett) [inline wolfHSM/src/chapter04.md:80]: source file links don't work - these probably shoudn't be links, lets just make them code

```suggestion
High-level client and server APIs (defined in `wolfhsm/wh_client.h` and `wolfhsm/wh_server.h` are the primary interface for communication. These functions abstract the low level communications details from the caller, providing a simple split transaction interface for logical operations.
```

File: wolfHSM/src/chapter04.md
Comment (bigbrett) [inline wolfHSM/src/chapter04.md:250]: broken links, markdown files were renamed

File: wolfHSM/src/chapter05.md
Comment (bigbrett) [inline wolfHSM/src/chapter05.md:3]: broken link

File: wolfHSM/src/chapter05.md
Comment (bigbrett) [inline wolfHSM/src/chapter05.md:199]: broken link - should point to DMA callbacks in chapter 7

File: wolfHSM/src/chapter05.md
Comment (bigbrett) [inline wolfHSM/src/chapter05.md:251]: broken link

File: wolfHSM/src/chapter05.md
Comment (bigbrett) [inline wolfHSM/src/chapter05.md:288]: broken link, but this section doesn't exist. I can add it in a PR later

File: wolfHSM/src/chapter07.md
Comment (bigbrett) [inline wolfHSM/src/chapter07.md:16]: broken link

File: wolfHSM/src/chapter08.md
Comment (bigbrett) [inline wolfHSM/src/chapter08.md:57]: POSIX should be all-caps, it is an acronym

**Review by bigbrett — CHANGES_REQUESTED** [review]
Awesome! First quick pass flagging some broken links, spelling errors, and a few suggestions.

**Comment by bigbrett** [comment]
@aidangarske You mentioned that we don't yet have doxygen - we do have the client and server APIs doxygen-commented, but it is inline in the source code (as opposed to maintained separately like wolfSSL does in this repo). There isnt a doxyfile yet though. Just FYI on where that is at

**Comment by dgarske** [comment]
> @aidangarske You mentioned that we don't yet have doxygen - we do have the client and server APIs doxygen-commented, but it is inline in the source code (as opposed to maintained separately like wolfSSL does in this repo). There isnt a doxyfile yet though. Just FYI on where that is at

Aidan is OOO today, but Ruby can help. Brett if you have API's that are public that you would like Doxygen to process please provide a link to those. Aidan and I could not find them.

```
/*!
    \ingroup Client Functions
    \brief Description of function

    \return 0=

    \param arg1 description 

    _Example_
    \code
    int rc = CallFunc();
    \endcode

    \sa OtherFunction
*/
```

Thanks

**Comment by bigbrett** [comment]
> > @aidangarske You mentioned that we don't yet have doxygen - we do have the client and server APIs doxygen-commented, but it is inline in the source code (as opposed to maintained separately like wolfSSL does in this repo). There isnt a doxyfile yet though. Just FYI on where that is at
> 
> @aidangarske is OOO today, but @rlm2002 can help. Brett if you have API's that are public that you would like Doxygen to process please provide a link to those. Aidan and I could not find them.
> 
> ```
> /*!
>     \ingroup Client Functions
>     \brief Description of function
> 
>     \return 0=
> 
>     \param arg1 description 
> 
>     _Example_
>     \code
>     int rc = CallFunc();
>     \endcode
> 
>     \sa OtherFunction
> */
> ```
> 
> Thanks

@rlm2002 here ya go:

- client API: https://github.com/wolfSSL/wolfHSM/blob/main/wolfhsm/wh_client.h
- server API: https://github.com/wolfSSL/wolfHSM/blob/main/wolfhsm/wh_server.h

**Review by JacobBarthelmeh — APPROVED** [review]

**Comment by JacobBarthelmeh** [comment]
Changes look good to me as a nice first step to adding the wolfHSM manual. The github actions is complaining about:

```
> [wolfssl-stage1 2/2] RUN make pdf V=:
0.992 [info] Loading docs/xml/wh__flash_8h.xml
0.992 [info] Loading docs/xml/wh__error_8h.xml
Warning: 0.993 [warning] Failed to parse member wh__error_8h error: basic_string: construction from null is not valid
0.993 [info] Loading docs/xml/wh__common_8h.xml
Warning: 0.994 [warning] Failed to parse member wh__common_8h error: basic_string: construction from null is not valid
0.994 [info] Loading docs/xml/README_8md.xml
```

**Comment by dgarske** [comment]
> Changes look good to me as a nice first step to adding the wolfHSM manual. The github actions is complaining about:
> 
> ```
> > [wolfssl-stage1 2/2] RUN make pdf V=:
> 0.992 [info] Loading docs/xml/wh__flash_8h.xml
> 0.992 [info] Loading docs/xml/wh__error_8h.xml
> Warning: 0.993 [warning] Failed to parse member wh__error_8h error: basic_string: construction from null is not valid
> 0.993 [info] Loading docs/xml/wh__common_8h.xml
> Warning: 0.994 [warning] Failed to parse member wh__common_8h error: basic_string: construction from null is not valid
> 0.994 [info] Loading docs/xml/README_8md.xml
> ```

Those are fixed with https://github.com/wolfSSL/wolfHSM/pull/42
Caused by unnamed enums and unions.
Once that PR is merged this should work. I've tested locally.

**Comment by JacobBarthelmeh** [comment]
@bigbrett it the request for changes still outstanding? Am okay merging in this initial documentation addition as is.

**Comment by bigbrett** [comment]
> @bigbrett is the request for changes still outstanding? Am okay merging in this initial documentation addition as is.

@JacobBarthelmeh let me take a peek. I'll approve if they are all addressed, otherwise I'll request changes.

**Review by bigbrett — APPROVED** [review]
@JacobBarthelmeh good to go.

@dgarske @aidangarske thank you both so much for your help on this. Go Team Garske!!!

@dgarske @JacobBarthelmeh Regarding the doxybook2 error, naming the enums is fine, but naming the unions will require changes to the calling code to work. We might end up doing that, or we may just pull docs into a separate header like wolfSSL does. Will discuss with the team. I don't think anonymous unions are critical to funcitonality, so we might just bite the bullet and refactor in this case. I'll keep you in the loop, but go ahead and merge for now without the API docs live.

**Pull request merged**
